OBAWidget/Provider/WidgetDataProvider.swift

```diff
@@ -10,48 +10,66 @@ import CoreLocation
 
 /// `WidgetDataProvider` is responsible for fetching and providing relevant data to the widget timeline provider.
 class WidgetDataProvider: NSObject, ObservableObject {
-    
-    public let formatters = Formatters(
-        locale: Locale.autoupdatingCurrent,
-        calendar: Calendar.autoupdatingCurrent,
-        themeColors: ThemeColors.shared
-    )
-    
     static let shared = WidgetDataProvider()
+
     private let userDefaults = UserDefaults(suiteName: Bundle.main.appGroup!)!
-    
     private lazy var locationManager = CLLocationManager()
     private lazy var locationService = LocationService(
         userDefaults: userDefaults,
         locationManager: locationManager
     )
-    
+
     private lazy var app: CoreApplication = {
-        let bundledRegions = Bundle.main.path(forResource: "regions", ofType: "json")!
-        let config = CoreAppConfig(appBundle: Bundle.main, userDefaults: userDefaults, bundledRegionsFilePath: bundledRegions)
+        let config = CoreAppConfig(
+            appBundle: Bundle.main,
+            userDefaults: userDefaults,
+            bundledRegionsFilePath: Bundle.main.path(forResource: "regions", ofType: "json")!
+        )
         return CoreApplication(config: config)
     }()
-    
+
     private var bestAvailableBookmarks: [Bookmark] {
         var bookmarks = app.userDataStore.favoritedBookmarks
         if bookmarks.isEmpty {
             bookmarks = app.userDataStore.bookmarks
         }
         return bookmarks
     }
-    
+
+    /// Dictionary mapping trip bookmark keys to arrival/departure data.
+    private var arrDepDic = [TripBookmarkKey: [ArrivalDeparture]]()
+
+    /// Formatters for localization and styling.
+    let formatters = Formatters(
+        locale: Locale.autoupdatingCurrent,
+        calendar: Calendar.autoupdatingCurrent,
+        themeColors: ThemeColors.shared
+    )
+
     /// Loads arrivals and departures for all favorited bookmarks for the widget.
-    public func loadData() async {
-        guard let apiService = app.apiService else { return }
-        
+    func loadData() async {
+        arrDepDic = [:] 
+
+        guard let apiService = app.getNewRefreshedRESTAPIService() else {
```

Comment:
call `app.refreshServices()` and then use `app.apiService`

Reply:
Done ✅